src/main/java/com/google/api/generator/gapic/composer/ClassComposer.java

```diff
@@ -16,9 +16,11 @@
 
 import com.google.api.generator.gapic.model.GapicClass;
 import com.google.api.generator.gapic.model.Message;
+import com.google.api.generator.gapic.model.ResourceName;
 import com.google.api.generator.gapic.model.Service;
 import java.util.Map;
 
 public interface ClassComposer {
-  GapicClass generate(Service service, Map<String, Message> messageTypes);
+  GapicClass generate(
+      Service service, Map<String, ResourceName> resourceNames, Map<String, Message> messageTypes);
```

Comment:
Can we avoid adding another parameter (that isn't used in most places) to this interface? I would prefer making the Composer (that actually uses this) not inherit from ClassComposer instead. Please add a TODO on me to clean up the hierarchy afterwards.

Reply:
Added. Thanks for doing that.

Reply:
To say this a different way: Could we avoid adding the `resourceNames` parameter to this interface? The TODO for me to clean this up would go into the classes that need the parameter, which would now no longer implement `ClassComposer`.